### ISSUE TITLE
Idea: instanceOf(containerKey)

### ISSUE THREAD

**Issue by dfreeman** [ciena-blueplanet/ember-prop-types#126]
I've had this idea in the back of my mind for a while now, and I wanted to run it here to get your thoughts before spending time on a PR. What do you think of augmenting `instanceOf` to allow it to take a container key?

This would allow something like this:
```js
// ...
import SomeModel from 'my-app/models/some-model';

export default Component.extend({
  propTypes: {
    prop: PropTypes.instanceOf(SomeModel)
  }
});
```

To become:
```js
// ...

export default Component.extend({
  propTypes: {
    prop: PropTypes.instanceOf('model:some-model')
  }
});
```

This seems (to me) to align nicely with how most other things in Ember work against the resolver rather than using direct imports. I did a quick pass over the existing implementation and I _believe_ this would be pretty simple to implement—if I opened a PR, would you be interested?

**Comment by sglanzer** [comment]
Yup, definitely interested - I believe @notmessenger was looking for something like this very recently
